daemon/runtime_unix.go

```diff
@@ -134,3 +137,35 @@ func (daemon *Daemon) getRuntime(name string) (*types.Runtime, error) {
 
 	return rt, nil
 }
+
+// isValidC8dRuntimeName tests whether name
+// is a well-formed ContainerD runtime name,
+// with additional Docker-specific restrictions.
+//
+// A runtime name containing slash characters is interpreted by ContainerD
+// as the path to a runtime binary.
+// If we allowed this, anyone with Engine API access could get ContainerD
+// to execute an arbitrary binary as root.
+// Although Engine API access is already equivalent to root on the host,
+// the runtime name has not historically been a vector
+// to run arbitrary code as root
+// so users are not expecting it to become one.
+//
+// This restriction is not configurable.
+// There are viable workarounds for legitimate use cases:
+// administrators and runtime developers can make runtimes available
+// for use with Docker by installing them onto $PATH
+// following the [binary naming convention] for ContainerD Runtime v2.
```

Comment:
I'm giving [semantic linefeeds](https://rhodesmill.org/brandon/2012/one-sentence-per-line/) (https://sembr.org/) a try. I'm interested to hear your thoughts and opinions on it.

Reply:
Hm... honestly; not a fan. I find it hard to read with the (for the reader) random line breaks.

I can see the advantage for _writing_ it (as well as in some cases to _update_), but (assuming the "happy path"), updates should be a lot less frequent than "reading". And while Markdown (probably as well when rendered as HTML on pkg.go.dev) would render it similar, I'm more often reading it as "source" than on the website 🤔

Reply:
I first learned about semantic linefeeds from [go.dev/doc/comment](https://go.dev/doc/comment#:~:text=The%20beginning%20of,when%20printing%20it.). It probably is overkill for unexported functions, and maybe even all commentary outside of packages intended for external users like `client` and `pkg/*` where the primary readers of the comments will likely be doing so from pkg.go.dev.